x-pack/plugins/enterprise_search/server/routes/enterprise_search/search_applications.ts

```diff
@@ -100,21 +101,45 @@ export function registerSearchApplicationsRoutes({ log, router }: RouteDependenc
 
         return response.ok({ body: engine });
       } catch (error) {
-        if (isVersionConflictEngineException(error)) {
-          return createError({
-            errorCode: ErrorCode.SEARCH_APPLICATION_ALREADY_EXISTS,
-            message: i18n.translate(
-              'xpack.enterpriseSearch.server.routes.createSearchApplication.searchApplciationExistsError',
-              {
-                defaultMessage: 'Search application name already taken. Choose another name.',
-              }
-            ),
-            response,
-            statusCode: 409,
-          });
+        switch (true) {
+          case isVersionConflictEngineException(error):
+            return createError({
+              errorCode: ErrorCode.SEARCH_APPLICATION_ALREADY_EXISTS,
+              message: i18n.translate(
+                'xpack.enterpriseSearch.server.routes.createSearchApplication.searchApplciationExistsError',
+                {
+                  defaultMessage: 'Search application name already taken. Choose another name.',
+                }
+              ),
+              response,
+              statusCode: 409,
+            });
+          case isInvalidSearchApplicationNameException(error):
+            let exceptionReason = '';
+            const unSupportedCharacters =
+              error.meta?.body?.error?.reason?.match(/\[(.*?)\]|'(.*?)'/gi);
+
+            if (unSupportedCharacters && unSupportedCharacters.length === 2) {
+              exceptionReason =
+                'Search application name must not contain: ' +
+                unSupportedCharacters[1].replace(/\'(.*?)\'/g, ' $& ');
```

Comment:
I found the error from Elasticsearch to be a bit unreadable in UI. Added this to create some spaces around each character

Reply:
Suggestion/Enhancement(non-blocking/please do as a separate PR): 

You can reuse of the validation from index creation FE. The rules are static so it could be shown as a better UI.

Reply:
Created a ticket to track the work: https://github.com/elastic/enterprise-search-team/issues/5051